terraform/modules/cluster/eks.tf

```diff
@@ -4,20 +4,38 @@ locals {
   default_mng_size = 2
 }
 
-module "eks_blueprints" {
-  source = "github.com/aws-ia/terraform-aws-eks-blueprints?ref=v4.16.0"
-
-  tags = local.tags
-
-  vpc_id             = module.aws_vpc.vpc_id
-  private_subnet_ids = local.private_subnet_ids
-  public_subnet_ids  = module.aws_vpc.public_subnets
+#tfsec:ignore:aws-eks-enable-control-plane-logging
+module "eks" {
+  source  = "terraform-aws-modules/eks/aws"
+  version = "~> 19.10"
+
+  cluster_name                   = var.environment_name
+  cluster_version                = var.cluster_version
+  cluster_endpoint_public_access = true
+
+  cluster_addons = {
+    coredns    = {}
+    kube-proxy = {}
+    vpc-cni = {
+      # Specify the VPC CNI addon should be deployed before compute to ensure
+      # the addon is configured before data plane compute resources are created
+      before_compute = true
+      most_recent    = true
+      configuration_values = jsonencode({
+        env = {
+          ENABLE_PREFIX_DELEGATION = "true"
+          ENABLE_POD_ENI           = "true"
+
+          POD_SECURITY_GROUP_ENFORCING_MODE = "standard"
+        }
+      })
+    }
+  }
 
-  cluster_name    = var.environment_name
-  cluster_version = var.cluster_version
+  kms_key_enable_default_policy = true
```

Comment:
replicates prior functionality to ensure no KMS key conflicts https://github.com/terraform-aws-modules/terraform-aws-kms/blob/87be9ccb61f073b58ed56de0456214c34d793841/main.tf#L89-L103

The KMS key created by the EKS module makes the cluster creator identity the key admin